server/internal/infrastructure/user/user_pg_test.go

```diff
@@ -21,8 +21,8 @@ func TestPgUsers_Count(t *testing.T) {
 
 		users := NewPgUsers(db)
 
-		testutil.Must(users.Save(ctx, testutil.Must(user.Create("a", "a"))))
-		testutil.Must(users.Save(ctx, testutil.Must(user.Create("b", "b"))))
+		testutil.Must(users.Save(ctx, *testutil.Must(user.Create("a", "a"))))
+		testutil.Must(users.Save(ctx, *testutil.Must(user.Create("b", "b"))))
```

Comment:
Is it really needed? `testutil.Must` returns T, so if `user.Create` returns `*User` we're already covered

Reply:
What do you mean?

Reply:
You added a pointer reference on the return type of `testutil.Must`, however `Must` is generic over `T` and returns `T`.

Unless generics in Go work wildly differently compared to other languages, that means that passing `*user.User` to `Must` returns ` *user.User`, right?

In that case you don't need to add another `*`

Reply:
`user.Create` returns `*User` and Must returns the same type. Save instead wants `User` so we need to use `*` again to point the content of the pointer

Reply:
Aaaaaah right I got it.
Would it make sense for `Save` to then accept a `*user.User`? Which one is more idiomatic?

Reply:
So, if the struct contains a lot of data you want to use the pointer to prevent it to be copied every time, in this case the user size is limited, so we can pass the argument without a pointer.

I personally prefer to pass always a struct without a pointer to avoid possible bugs if no validation is applied on `nil`, the idea is to move up in the stack trace the validation `if user != nil then use *user` but of course it has a cost in memory (negligible this time).

Up to you, for me it's the same

Reply:
Let's keep it like it is then